src/storage/src/table/merge_sort.rs

```diff
@@ -18,47 +18,47 @@ use std::error::Error;
 
 use futures::{Stream, StreamExt};
 use futures_async_stream::try_stream;
-use risingwave_common::row::OwnedRow;
 
 pub trait MergeSortKey = Eq + PartialEq + Ord + PartialOrd;
 
-struct Node<K: MergeSortKey, S> {
+struct Node<K: MergeSortKey, V, S> {
     stream: S,
 
     /// The next item polled from `stream` previously. Since the `eq` and `cmp` must be synchronous
     /// functions, we need to implement peeking manually.
-    peeked: (K, OwnedRow),
+    peeked: (K, V),
```

Comment:
The `K` here now will be `(VirtualNode, Bytes)` but I think what we need is the `Bytes` part.

Reply:
Since `VirtualNode` is passed to `iter_xx_with_pk_range` AP, I thought it will be helpful to expose vnode as part of the key to the upper layer as well and upper layer can decide whether they want to discard it or not.

Reply:
If we can confirm that vnode is not needed in the returning key, I can change that as well.

Reply:
I prefer to keep existing api in this PR as is for flexibility. Some components will still want to iter with vnode, e.g. arrangement backfill.

Introducing a new iter which just drops the vnode part will be good enough IMO.